controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -1262,3 +1263,104 @@ def test_identity_probe(self):
         request = Mock()
         context = Mock()
         self.servicer.Probe(request, context)
+
+
+class TestControllerServerValidateVolumeCapabilities(AbstractControllerTest):
+
+    def get_create_object_response_method(self):
+        return csi_pb2.ValidateVolumeCapabilitiesResponse
+
+    def get_create_object_method(self):
+        return self.servicer.ValidateVolumeCapabilities
+
+    @patch("controller.array_action.array_mediator_xiv.XIVArrayMediator._connect")
+    def setUp(self, connect):
+        self.fqdn = "fqdn"
+        self.hostname = "hostname"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.mediator.get_volume_by_id = Mock()
+        self.mediator.get_volume_by_id.return_value = utils.get_mock_mediator_response_volume(10, "vol", "wwn2", "a9k")
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
+        self.request = Mock()
+        arr_type = XIVArrayMediator.array_type
+        self.request.volume_id = "{}:wwn1".format(arr_type)
+        self.request.secrets = {"username": "user", "password": "pass", "management_address": "mg"}
+        self.request.parameters = {}
+        self.request.volume_context = {}
+        caps = utils.get_mock_volume_capability_object()
+        self.request.volume_capabilities = [caps]
+
+        self.context = utils.FakeContext()
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_success(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+
+        self.servicer.ValidateVolumeCapabilities(self.request, self.context)
+
+        self.assertEqual(self.context.code, grpc.StatusCode.OK)
+
+    @patch("controller.controller_server.csi_controller_server.get_agent")
+    def test_validate_volume_cap_with_empty_id(self, storage_agent):
+        storage_agent.return_value = self.storage_agent
+        self.request.volume_id = ""
+        res = self.get_create_object_method()(self.request, self.context)
+        self.assertEqual(self.context.code, grpc.StatusCode.INVALID_ARGUMENT)
+        self.assertTrue("volume id" in self.context.details)
+        self.assertEqual(res, self.get_create_object_response_method()(message=self.context.details))

```

Comment:
avoid multiple calls in 1 line, especially [train wrecks](https://wiki.c2.com/?TrainWreck)

Reply:
done

Reply:
I don't see any change here

Reply:
now, done

Reply:
thanks. [this](https://github.com/IBM/ibm-block-csi-driver/pull/212#discussion_r517928021) could make it even better